srtcore/srt.h

```diff
@@ -757,15 +757,37 @@ SRT_ATR_DEPRECATED_PX SRT_API SRTSOCKET srt_socket(int, int, int) SRT_ATR_DEPREC
 SRT_API       SRTSOCKET srt_create_socket();
 
 // Group management
+
+
+typedef enum SRT_MemberStatus
+{
+    SRT_GST_PENDING,  // The socket is created correctly, but not yet ready for getting data.
+    SRT_GST_IDLE,     // The socket is ready to be activated
+    SRT_GST_RUNNING,  // The socket was already activated and is in use
+    SRT_GST_BROKEN    // The last operation broke the socket, it should be closed.
+} SRT_MEMBERSTATUS;
+
+
 typedef struct SRT_SocketGroupData_
 {
     SRTSOCKET id;
-    SRT_SOCKSTATUS status;
+    struct sockaddr_storage peeraddr; // Don't want to expose sockaddr_any to public API
+    SRT_SOCKSTATUS sockstate;
+    SRT_MEMBERSTATUS memberstate;
```

Comment:
`Status state`.
Maybe `SRT_SOCKSTATUS sockstatus;` and `SRT_MEMBERSTATUS memberstatus;` then?

Reply:
The tricky part is that the function to obtain it for a single socket is `srt_getsockstate`. And yes, it returns the `SRT_SOCKSTATUS` type.